reports/report.md

```diff
@@ -1143,7 +1145,96 @@ If the length of a dynamic array (storage variable) directly assigned to, it may
 
 
 
-## H-16: Yul block contains `return` function call.
+## H-16: Uninitialized State Variables
+
+Solidity does initialize variables by default when you declare them, however it's good practice to explicitly declare an initial value. For example, if you transfer money to an address we must make sure that the address has been initialized.
+
+<details><summary>13 Found Instances</summary>
+
+
+- Found in src/AssemblyExample.sol [Line: 5](../tests/contract-playground/src/AssemblyExample.sol#L5)
+
+	```solidity
+	    uint b;
+	```
+
+- Found in src/InconsistentUints.sol [Line: 7](../tests/contract-playground/src/InconsistentUints.sol#L7)
+
+	```solidity
+	    int public intVariable; // 1
+	```
+
+- Found in src/InconsistentUints.sol [Line: 8](../tests/contract-playground/src/InconsistentUints.sol#L8)
+
+	```solidity
+	    int256 public int256Variable; // 1 
+	```
+
+- Found in src/StateVariables.sol [Line: 8](../tests/contract-playground/src/StateVariables.sol#L8)
+
+	```solidity
+	    uint256 private staticPrivateNumber;
+	```
+
+- Found in src/StateVariables.sol [Line: 9](../tests/contract-playground/src/StateVariables.sol#L9)
+
+	```solidity
+	    uint256 internal staticInternalNumber;
+	```
+
+- Found in src/StateVariables.sol [Line: 10](../tests/contract-playground/src/StateVariables.sol#L10)
+
+	```solidity
+	    uint256 public staticPublicNumber;
+	```
+
+- Found in src/StorageParameters.sol [Line: 5](../tests/contract-playground/src/StorageParameters.sol#L5)
+
+	```solidity
+	    uint[1] public storageArray;
+	```
+
+- Found in src/UninitializedStateVariable.sol [Line: 7](../tests/contract-playground/src/UninitializedStateVariable.sol#L7)
+
+	```solidity
+	    string public s_author; // BAD (because it's used without initializing)
+	```
+
+- Found in src/UninitializedStateVariable.sol [Line: 15](../tests/contract-playground/src/UninitializedStateVariable.sol#L15)
+
+	```solidity
+	    address destination; // BAD
+	```
+
+- Found in src/WrongOrderOfLayout.sol [Line: 11](../tests/contract-playground/src/WrongOrderOfLayout.sol#L11)
+
+	```solidity
+	    uint256 public multiplier;
+	```
+
+- Found in src/auditor_mode/PublicFunctionsWithoutSenderCheck.sol [Line: 68](../tests/contract-playground/src/auditor_mode/PublicFunctionsWithoutSenderCheck.sol#L68)
+
+	```solidity
+	    address public owner;
+	```
+
+- Found in src/eth2/DepositContract.sol [Line: 63](../tests/contract-playground/src/eth2/DepositContract.sol#L63)
+
+	```solidity
+	    bytes32[DEPOSIT_CONTRACT_TREE_DEPTH] branch;
+	```
+
+- Found in src/eth2/DepositContract.sol [Line: 66](../tests/contract-playground/src/eth2/DepositContract.sol#L66)
+
+	```solidity
+	    bytes32[DEPOSIT_CONTRACT_TREE_DEPTH] zero_hashes;
```

Comment:
This seems to catch an array that actually is assigned in the constructor. Is this an edge case not accounted for?

Reply:
Good catch! This is a fixed size array. It also needs to be accounted for

Reply:
✅